go-controller/pkg/ovn/secondary_layer2_network_controller.go

```diff
@@ -615,3 +680,30 @@ func (oc *SecondaryLayer2NetworkController) gatewayManagerForNode(nodeName strin
 		return gwManager
 	}
 }
+
+func (oc *SecondaryLayer2NetworkController) gatewayOptions() []GatewayOption {
+	var opts []GatewayOption
+	if oc.clusterLoadBalancerGroupUUID != "" {
+		opts = append(opts, WithLoadBalancerGroups(
+			oc.routerLoadBalancerGroupUUID,
+			oc.clusterLoadBalancerGroupUUID,
+			oc.switchLoadBalancerGroupUUID,
+		))
+	}
+	return opts
+}
+
+func (oc *SecondaryLayer2NetworkController) StartServiceController(wg *sync.WaitGroup, runRepair bool) error {
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		useLBGroups := oc.clusterLoadBalancerGroupUUID != ""
+		// use 5 workers like most of the kubernetes controllers in the
+		// kubernetes controller-manager
+		err := oc.svcController.Run(5, oc.stopChan, runRepair, useLBGroups, oc.svcTemplateSupport)
+		if err != nil {
+			klog.Errorf("Error running OVN Kubernetes Services controller: %v", err)
```

Comment:
would it be good to list the name of the network that the service controller couldn't start for?